crates/flux-refineck/src/primops.rs

```diff
@@ -365,13 +367,16 @@ fn mk_neg_rules(check_overflow: bool) -> RuleMatcher<1> {
             fn(a: T) -> T[a.neg()]
             requires E::ne(a, E::int_min(int_ty)) => ConstrReason::Overflow
             if let &BaseTy::Int(int_ty) = T
+
+            fn(a: T) -> T[a.neg()]
+            if T.is_float()
```

Comment:
Floats are not (currently) refined, so this generates a malformed constraint.

```suggestion
            fn(a: T) -> T
            if T.is_float()
```